command/setup_consul.go

```diff
@@ -0,0 +1,449 @@
+  # CONSUL_HTTP_TOKEN environment variable instead of writing it in the
+  # configuration file.
+
+  # Consul mTLS configuration.
+  # ssl       = true
+  # ca_file   = "/var/ssl/bundle/ca.bundle"
+  # cert_file = "/etc/ssl/consul.crt"
+  # key_file  = "/etc/ssl/consul.key"
+
+  service_identity {
+    aud = ["consul.io"]
+    ttl = "1h"
+  }
+
+  task_identity {
+    aud = ["consul.io"]
+    ttl = "1h"
+  }
+}
+
+`)
+
+	return 0
+}
+
+func (s *SetupConsulCommand) renderAuthMethodConf(authMethodName string) (map[string]any, error) {
+	authConfig := map[string]any{}
+	err := json.Unmarshal(authConfigBody, &authConfig)
+	if err != nil {
+		return authConfig, fmt.Errorf("default auth config text could not be deserialized: %v", err)
+	}
+
+	authConfig["JWKSURL"] = s.jwksURL
+	authConfig["BoundAudiences"] = []string{aud}
+	authConfig["JWTSupportedAlgs"] = []string{"RS256"}
+
+	return authConfig, nil
+}
+
+func (s *SetupConsulCommand) createAuthMethod(authMethodName string, authMethodConf map[string]any) error {
+	_, _, err := s.client.ACL().AuthMethodCreate(&api.ACLAuthMethod{
+		Name:          authMethodName,
+		Type:          "jwt",
+		DisplayName:   authMethodName,
+		Description:   "login method for Nomad workload identities (WI)",
+		MaxTokenTTL:   defaultTTL,
+		TokenLocality: "local",
+		Config:        authMethodConf,
+		NamespaceRules: []*api.ACLAuthMethodNamespaceRule{{
+			Selector:      "",
+			BindNamespace: "${value.nomad_namespace}",
+		}},
+	}, nil)
+
+	if err != nil {
+		return fmt.Errorf("[✘] could not create Consul auth method: %w", err)
+	}
+
+	s.Ui.Info(fmt.Sprintf("[✔] Created auth method %s", authMethodName))
+	return nil
+}
+
+func (s *SetupConsulCommand) createBindingRules(rule *api.ACLBindingRule) error {
+	_, _, err := s.client.ACL().BindingRuleCreate(rule, nil)
+	if err != nil {
+		return fmt.Errorf("[✘] could not create Consul binding rule: %w", err)
+	}
+
+	s.Ui.Info(fmt.Sprintf("[✔] Created binding rule for auth method %s", rule.AuthMethod))
+	return nil
+}
+
+func (s *SetupConsulCommand) createRoleForTemplate() error {
+	_, _, err := s.client.ACL().RoleCreate(&api.ACLRole{
+		Name:        roleTasks,
+		Description: "role for Nomad templates w/ workload identities (WI)",
+		Policies:    []*api.ACLLink{{Name: policyName}},
+	}, nil)
+	if err != nil {
+		if strings.Contains(err.Error(), "already exists") {
+			s.Ui.Warn(fmt.Sprintf("[ ] role %s already exists", roleTasks))
+			return nil
+		}
+		return fmt.Errorf("[✘] could not create Consul role: %w", err)
+	}
+
+	s.Ui.Info(fmt.Sprintf("[✔] Created role %s\n", roleTasks))
+	return nil
+}
+
+func (s *SetupConsulCommand) createPolicy() error {
+	_, _, err := s.client.ACL().PolicyCreate(&api.ACLPolicy{
+		Name:  policyName,
+		Rules: string(policyBody),
+	}, nil)
+	if err != nil {
+		if strings.Contains(err.Error(), "already exists") {
+			s.Ui.Warn(fmt.Sprintf("[ ] policy %s already exists", policyName))
+		}
```

Comment:
Do we want to return early here like it's being done in the role function above?